web/src/main/java/org/cbioportal/web/util/LinearDataBinner.java

```diff
@@ -36,26 +36,44 @@ public LinearDataBinner(DataBinHelper dataBinHelper) {
                                            Range<Double> boxRange,
                                            List<Double> values,
                                            Double lowerOutlier,
-                                           Double upperOutlier)
-    {
+                                           Double upperOutlier) {
         Double min = lowerOutlier == null ? Collections.min(values) : Math.max(Collections.min(values), lowerOutlier);
         Double max = upperOutlier == null ? Collections.max(values) : Math.min(Collections.max(values), upperOutlier);
-        
-                
+
         List<DataBin> dataBins = initDataBins(attributeId, min, max, lowerOutlier, upperOutlier);
 
         // special case for "AGE" attributes
         if (dataBinHelper.isAgeAttribute(attributeId) &&
```

Comment:
Not sure if I got it right, but it seems like you added both custom bin list and custom range support, so does this mean now frontend can check if it's an age attribute and send a custom range?

Reply:
Correct, just validate that. Will update the test and remove the special condition for AGE